plasmapy/utils/pytest_helpers.py

```diff
@@ -93,35 +207,134 @@ def run_test(func: Callable, args: Any = (), kwargs: Dict = {}, expected_outcome
+
     Examples
     --------
+    The simplest way to use `~plasmapy.utils.run_test` is with inputs
+    for the function to be tests, the positional arguments in a `tuple`
+    or `list`, the keyword arguments in a `dict`, and then finally the
+    expected result or outcome.
+
+    >>> args = tuple()
+    >>> kwargs = dict()
+    >>> run_test(lambda: 0, args, kwargs, 0)
+
+    If `expected` is a an exception or warning, then
+    `~plasmapy.atomic.run_test` will raise an exception if the expected
+    exception is not raised or the expected warning is not issued.
+
     >>> from warnings import warn
-    >>> run_test(lambda: 42, tuple(), dict(), 42)
-    >>> run_test(lambda: warn("", UserWarning), tuple(), dict(), UserWarning)
 
-    """
+    >>> issue_warning = lambda: warn("Electrons are weird!", UserWarning)
+    >>> run_test(issue_warning, args, kwargs, UserWarning)
+
+    >>> def raise_exception(): raise RuntimeError
+    >>> run_test(raise_exception, args, kwargs, RuntimeError)
+
+    For warnings, `~plasmapy.utils.run_test` can accept a `tuple` of two
+    items where the first item is the expected result and the second
+    item is the expected warning.
+
+    .. code-block:: python
+
+        def return_arg_and_warn(x):
+            warn("", UserWarning)
+            return x
+
+        run_test(return_arg_and_warn, 1, {}, (1, UserWarning))
+
+    This function is also flexible enough that it can accept a `tuple`
+    or `list` as its sole argument, with the arguments in the same
+    order as in the function signature.
+
+    >>> return_arg = lambda x: x
+    >>> inputs = (return_arg, 42, {}, 42)
+    >>> run_test(inputs)
+
+    If the `tuple` or `list` has a length of `3`, then
+    `~plasmapy.utils.run_test` assumes that `kwargs` is missing.
+
+    >>> inputs_without_kwargs = [return_arg, 42, 42]
+    >>> run_test(inputs_without_kwargs)
+
+    .. code-block:: python
+
+        import pytest
 
-    def exc_str(ex: Exception) -> str:
-        """
-        Return a string with an indefinite article and the name of
-        exception `ex`.
-        """
-        article = 'an' if ex.__name__[0] in 'aeiouAEIOU' else 'a'
-        return f"{article} {ex.__name__}"
+        def func(x, raise_exception=False, issue_warning=False):
+            if raise_exception:
+                raise ValueError("I'm sorry, Dave. I'm afraid I can't do that.")
+            elif issue_warning:
+                warn("Open the pod bay doors, HAL.", UserWarning)
+            return x
+
+        inputs_table = [
+            (func, 1, 1),
+            (func, (2,), {}, 2),
+            (func, 3, {'raise_exception': True}, ValueError),
+            (func, 4, {'issue_warning': True}, UserWarning),
+            (func, 5, {'issue_warning': True}, (5, UserWarning)),
+        ]
+
+        @pytest.mark.parametrize('inputs', inputs_table)
+        def test_func(inputs):
+            run_test(inputs)
+
+    """
```

Comment:
This example is pretty darn long. Do you think it would be helpful to let people play with it via jupyter notebook?